txscript/script_test.go

```diff
@@ -540,3 +534,14 @@ func TestIsUnspendable(t *testing.T) {
 		}
 	}
 }
+
+// TestTreasury verifies that treasury scripts are properly detected.
+func TestTreasury(t *testing.T) {
+	script, err := PayToTreasury()
```

Comment:
I removed this since this function is not going to be used. I thought the wallet would use it but it's silly since it only adds an OP_TADD and the rest is done in the wallet.